fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java

```diff
@@ -1857,15 +1898,37 @@ public NormalBackendSelector(ScanNode scanNode, List<TScanRangeLocations> locati
             this.assignment = assignment;
         }
 
+        public NormalBackendSelector(ScanNode scanNode, List<TScanRangeLocations> locations,
+                                     FragmentScanRangeAssignment assignment, boolean isLoad) {
+            this.scanNode = scanNode;
+            this.locations = locations;
+            this.assignment = assignment;
+            this.isLoad = isLoad;
+        }
+
         @Override
         public void computeScanRangeAssignment() throws Exception {
-            HashMap<TNetworkAddress, Long> assignedBytesPerHost = Maps.newHashMap();
+            HashMap<TNetworkAddress, Long> assignedRowCountPerHost = Maps.newHashMap();
+            // sort the scan ranges by row count
+            // only sort the scan range when it is load job
+            // but when there are too many scan ranges, we will not sort them since performance issue
+            if (Config.enable_schedule_insert_query_by_row_count && isLoad
+                    && locations.size() < 10240 && locations.size() > 0
+                    && locations.get(0).getScan_range().isSetInternal_scan_range()) {
```

Comment:
How about abstracting the condition to a variable, since it is too long?

```suggestion
            boolean needSortLocations = Config.enable_schedule_insert_query_by_row_count && isLoad
                    && locations.size() < 10240 && locations.size() > 0
                    && locations.get(0).getScan_range().isSetInternal_scan_range();
            if (needSortLocations) {
```